DeveloperGuide: remove all references to `BrowserPanel`

pyokagan opened the pull request:
With the removal of the browser panel in 1e8e8dfc (Remove BrowserPanel,
2019-01-25), `BrowserPanel` does not exist any more.

damithc commented:
@YijinL can you check and merge?
Sorry, your latest commit (adding comments) got wiped out.

Things to note:
* I swapped the the parameters in AssertEquals calls. It should be (expected, actual)
* Replaces AssertEquals(..., true) with AssertTrue(...)
* Some repetitive code blocks remain, but this will have to do for now.

pyokagan committed:
DeveloperGuide: remove all references to `BrowserPanel`

With the removal of the browser panel in 1e8e8dfc (Remove BrowserPanel,
2019-01-25), `BrowserPanel` does not exist any more.
